repos/system_upgrade/common/actors/livemode/livemode_config_scanner/libraries/scan_livemode_config.py

```diff
@@ -0,0 +1,118 @@
+import configparser
+
+from leapp.libraries.common.config import architecture, get_env
+from leapp.libraries.common.rpms import has_package
+from leapp.exceptions import StopActorExecutionError
+from leapp.models import InstalledRPM, LiveModeConfig
+from leapp.models.fields import ModelViolationError
+from leapp.libraries.stdlib import api
+
+
+LIVEMODE_CONFIG_LOCATION = '/etc/leapp/files/devel-livemode.ini'
+DEFAULT_SQUASHFS_PATH = '/var/lib/leapp/live-upgrade.img'
+
+
+def scan_config_and_emit_message():
```

Comment:
You are right, the function does do a lot of actions - I would say that the logic of whether or not should it run could be separated.

On a more general note - I don't like splitting functions just because it does more than one action. Surely, it is hard to unit-test the function, but it can be much easier to debug it (not dealing with complex call-graph) as one sees the code that is executed in its entirety and can simply step through it (even mentally). However, as python does not allow the programmer to distinguish a block of code as one can do e.g. in C++, introducing a function is the only mechanism for grouping together lines of code that would like to achieve something while exposing the action (i.e. function name).

Reply:
The reason to split functions is actually to make it easier to read and debug.  But, depending on how the code was divided into subfunctions, it can make it easier or harder.

The idea behind making it easier is that you can only hold so much information in your active focus at one time.  If all of the code is in a single function, then you have to examine and hold a lot of the information about all of the code in your mind simultaneously.  In these cases, a well-named function allows you to understand what the function does at a higher level and only dive into the specifics when you suspect that the bug is inside of a specific code path.

Problems arise when people don't really understand how to split the code they write into functions which can achieve this.  Poor naming, passing too many variables, returning multiple data types in different circumstances, a lack of organization of the calling code into distinct steps, sub-functions that have too many different behaviours depending on the circumstances, etc can all mean that someone has to read the sub-function in order to understand and debug the higher level function.  When that happens, the purpose of dividing the code into functions has been lost.

The goal of good use of subfunctions is to:
* Be able to read the code and understand what it is intended to do without having read the implementation of the subfunctions.
* Be able to look at your error and know which subfunction is a good place to start debugging.
* To only have to step through the code that's relevant to your problem and are able to ignore the rest (this is the ideal, normally, you have a subset of the subfunctions where the problem could exist but you can either take a good guess as to the most likely places or poke at the data you have between calls to functions to bisect your way to the problem).